library/src/main/java/com/wultra/android/mtokensdk/operation/OperationsService.kt

```diff
@@ -353,3 +363,60 @@ class OperationsService: IOperationsService {
         Logger.d("Operation polling stopped")
     }
 }
+
+data class CallbackData(val operationsList: List<UserOperation>, val removed: List<UserOperation>, val added: List<UserOperation>)
+
+private class OperationsRegister(private val onChangeCallback: (CallbackData) -> Unit) {
+    private var currentOperations = mutableListOf<UserOperation>()
+    private var currentOperationIds = mutableSetOf<String>()
+
+    // Adds an operation to the register
+    fun add(operation: UserOperation) {
+        if (!currentOperations.any { it.id == operation.id }) {
```

Comment:
The `add`, `replace` and `remove` methods should be synchronized to prevent race conditions (ideally by using mutex like in the service).

Imagine that the operation update and claim are running at the same time - this could lead to a completely unexpected listener reporting.